### PR TITLE
Make 1Password util for Airflow 2 DAGs to grab secrets

### PR DESCRIPTION
**Pull request by mddilley**
## Associated issues
https://github.com/cityofaustin/atd-data-tech/issues/12432

This PR adds a 1Password utility to reduce lib imports and config in DAGs. I added a bit of info on the utils so far. Open to any feedback on the format of that info. I know the 1PW discussion is on-going. 🙏

## Testing

1. [Spin up the local stack
](https://github.com/cityofaustin/atd-airflow/blob/airflow-v2/README.md#local-setup)
2. This is working if the service bot DAGs are visible in the Airflow dashboard and there are no DAG import errors.

### CODE REVIEW

File: dags/utils/onepassword.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+def load_dict(required_secrets):
+    """Return dict of specified keys and secrets from 1Password vault.
+
+    Keyword arguments:
+    required_secrets -- a dict of specified keys and values of location config w/o vault id
+    """
+    for value in required_secrets.values():
+        value["opvault"] = VAULT_ID
+
+    return onepasswordconnectsdk.load_dict(client, REQUIRED_SECRETS)
```
Comment (frankhereford) [inline dags/utils/onepassword.py]: I think this line needs to be changed to reference the lower-case variable name `requiered_secrets`.

**Review by frankhereford — CHANGES_REQUESTED** [review]
I had to adjust a line to get this working, but otherwise, this is awesome!! Thank you Mike!

File: dags/utils/onepassword.py (hunk 1 above)
Comment (mddilley) [reply dags/utils/onepassword.py]: you are a lifesaver 🛟 - yep, this was a last minute change. Just fixed it. 🙏

**Review by frankhereford — APPROVED** [review]
So perfect - thanks!

**Review by chiaberry — APPROVED** [review]
This looks great

File: dags/utils/onepassword.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import onepasswordconnectsdk
+
+ONEPASSWORD_CONNECT_HOST = os.getenv("OP_CONNECT")
+ONEPASSWORD_CONNECT_TOKEN = os.getenv("OP_API_TOKEN")
+VAULT_ID = os.getenv("OP_VAULT_ID")
+
+client: Client = new_client(ONEPASSWORD_CONNECT_HOST, ONEPASSWORD_CONNECT_TOKEN)
+
+
+def load_dict(required_secrets):
```
Comment (johnclary) [inline dags/utils/onepassword.py:13]: i was initially going to suggest you make this function name more clear, like `load_secrets` - but i see you're proxying the `op` API: nice! 👍 🚢

**Review by johnclary — APPROVED** [review]
hooray for nixing boilerplate. nice work! 🚢🚢🚢🚢

**Pull request merged**